### ISSUE TITLE
how to set alpha for current holder as 1 and for others as 0.5 

### ISSUE THREAD

**Issue by Rohit3901** [yarolegovich/DiscreteScrollView#124]

**Comment by Rohit3901** [comment]
The problem I am facing is after scrolling the alpha of the previous holder doesn't change back to 0.5f

**Comment by Rohit3901** [comment]
Nope not yet

**Comment by joshthor3222** [comment]
OK. I think i have a solution for this. it seems to be what i was trying to do and I couldn't find anyone who did it with this library. 

Here is what I did:
1. Inside my card xml file I added a id to the cardview and set the default alpha to .6 - this will be my default alpha value for all background cards.
2. Inside my custom adapter i expanded the ViewHolder class. In the class I added a private CardView value, and inside the constructor I set it by finding it by the id I added. Additionally, I added a public method called "setAlpha" that takes a float that simply calls card.setAlpha with the new float. this will allow you to set the alpha whenever you feel like it.
3. In my activity I implemented DiscreteScrollView.ScrollStateChangeListener<ScrollAdapter.ViewHolder>where ScrollAdapter is my custom adapter. I overwrote "onScrollStart" and "onScrollEnd" with nothing. these are required to be overwritten by the Interface however. the real meat of the code is done by overwriting the "onScroll" class. my code is as follows: 

`public void onScroll(float position, int currentIndex, int newIndex,@Nullable ScrollAdapter.ViewHolder currentHolder, @Nullable ScrollAdapter.ViewHolder newHolder) {

        if(position < 0) {
            float newalpha = 1 - (float) ((1 + position) * .4);
            float oldalpha = 0 - (float) (position * .4);
            currentHolder.setAlpha(1 - oldalpha);
            newHolder.setAlpha(newalpha);
        }
        else {
            float currentAlpha = 1 - (float)(position * .4);
            float newAlpha = 0 - (float)((1 - position) * .4);
            currentHolder.setAlpha(currentAlpha);
            newHolder.setAlpha(1 + newAlpha);
        }
    }`

What this does is it uses the position vs the max and min opacity (1 and .6) to set the opacity as the position changes so it is linear and not just "front item gets alpha 1, back items get alpha .6"

4. The last step is in my custom adapter I set the initial item to have opacity 1. 

Inside the "onCreateViewHolder" method I added a bunch of code. we grab the parent context and see if a card exists inside it. if it does not we mark a bool labeled "first" as true. then after we inflate the view, we set the alpha on that view at 1.

`public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {

        boolean first = false;
        LayoutInflater inflater = LayoutInflater.from(parent.getContext());

        Context parentContext = parent.getContext();
        CardView card = (CardView) ((Activity) parentContext).findViewById(R.id.card);
        if(card == null) {
            first = true;
        }
        View v = inflater.inflate(R.layout.item_card, parent, false);

        if(first == true) {
            v.setAlpha(1);
        }
        return new ViewHolder(v);
    }`


Hopefully this helps!
